src/anticipate/decorators.py

```diff
@@ -22,40 +20,39 @@ def __init__(self, func, returns, params, strict=False):
         self.func = func
         self.returns = returns
         self.params = params
-        self.adapt = adapt
+        self._adapt_result = self._get_adapter(self.returns) if self.returns else None
         self.strict = strict
 
-        if isinstance(self.returns, list):
-            # Return value is a list of items matching the first element
-            if len(self.returns) == 1:
-                self.returns = self.returns[0]
-                self.adapt = adapt_all
-            else:
-                raise TypeError('An anticipated list can only contain one type')
+        args, _, kwargs, _ = inspect.getargspec(func)
+
+        self.arg_names = args
+
+        if not kwargs:
+            # If kwargs are accepted then any parameter name can be used.
+            # Otherwise, we check to see if there are parameters that do not
+            # match the funtion signature. This is a safety precaution to
+            # protect againts typo in parameter names.
+            invalid_params = set(self.params.keys()) - set(self.arg_names)
+            if invalid_params:
+                raise KeyError(
+                    'Invalid anticipate parameters found that do not match '
```

Comment:
Yeah, this one was bugging me and helped spot some typos in a large code base that uses anticipate.